torch/csrc/jit/script/parser.cpp

```diff
@@ -549,6 +549,23 @@ struct ParserImpl {
         paramlist.range(), List<Param>(paramlist), return_annotation);
   }
 
+  TreeRef parseClass() {
+    L.expect(TK_CLASS_DEF);
+    const auto name = parseIdent();
+    // TODO no inheritance or () allowed right now
+    L.expect(':');
+
+    L.expect(TK_INDENT);
+    std::vector<Def> methods;
+    while (L.cur().kind != TK_DEDENT) {
+      methods.push_back(Def(parseFunction(/*is_method=*/true)));
```

Comment:
Not sure that this would actually come up, but something like:
```
        class Hello:
            def foo():
                print("hi")

            def __init__(self):
                self.a = 1

```

Will that parse foo as a class method ?

Reply:
It could come up, but it should be the compilers job to reject it I think. The parser just defines the syntactically valid input language.